install/src/Stack.hs

```diff
@@ -0,0 +1,98 @@
+module Stack where
+
+import           Development.Shake
+import           Development.Shake.Command
+import           Development.Shake.FilePath
+import           Control.Monad
+import           System.Directory                         ( copyFile )
+
+import           Version
+import           Print
+import           Env
+
+
+stackBuildHie :: VersionNumber -> Action ()
+stackBuildHie versionNumber = execStackWithGhc_ versionNumber ["build"]
+  `actionOnException` liftIO (putStrLn stackBuildFailMsg)
+
+-- | copy the built binaries into the localBinDir
+stackInstallHie :: VersionNumber -> Action ()
+stackInstallHie versionNumber = do
+  execStackWithGhc_ versionNumber ["install"]
+  localBinDir <- getLocalBin
+  let hie = "hie" <.> exe
+  liftIO $ do
+    copyFile (localBinDir </> hie)
+             (localBinDir </> "hie-" ++ versionNumber <.> exe)
+    copyFile (localBinDir </> hie)
+             (localBinDir </> "hie-" ++ dropExtension versionNumber <.> exe)
+
+buildCopyCompilerTool :: VersionNumber -> Action ()
+buildCopyCompilerTool versionNumber =
+  execStackWithGhc_ versionNumber ["build", "--copy-compiler-tool"]
+
+
+-- | check `stack` has the required version
+checkStack :: Action ()
+checkStack = do
+  stackVersion <- trimmedStdout <$> execStackShake ["--numeric-version"]
+  unless (checkVersion requiredStackVersion stackVersion) $ do
+    printInStars $ stackExeIsOldFailMsg stackVersion
+    error $ stackExeIsOldFailMsg stackVersion
+
+-- | Get the local binary path of stack.
+-- Equal to the command `stack path --local-bin`
+getLocalBin :: Action FilePath
+getLocalBin = do
+  Stdout stackLocalDir' <- execStackShake ["path", "--local-bin"]
+  return $ trim stackLocalDir'
+
+stackBuildData :: Action ()
```

Comment:
-> `stackBuildHoogleDatabase` or something like that?
data is way more general than it actually is.

Reply:
When other data (e.g. `hlint`'s data-files) is required, this function will do that as well. I would keep the name identical to the target and I dont want to change the name of the target